MINI_2016/Homeworks/Fifth/h5_mvb.Rmd

```diff
@@ -0,0 +1,196 @@
+The newly created variables will be:
+
+- **BoughtProducts** - bought products per VisitNumber
+
+- **ReturnedProducts** - returned products per VisitNumber
+
+- **TotalProducts** - total products per VisitNumber
+
+- **ReturnedProducts** - returned products per VisitNumber
+
+- **DepartmentCount** - number of departments per VisitNumber
+
+- **FinelineCount** - number of finelines per VisitNumber
+
+- **BoughtPerDepartment** - returned products per Department
+
+- **ReturnedPerDepartment** - returned products per Department
+
+And also I will create features by spreading the `DepartmentDescription`.
+```{r, echo=TRUE, message=FALSE, warning=FALSE}
+data = import %>% 
+  group_by(VisitNumber) %>%
+  summarise(TripType = head(TripType,1),
+            Day = unique(setNames(0:6, c("Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"))[Weekday])[1],
+            BoughtProducts = sum(ScanCount >= 0),
+            ReturnedProducts = sum(ScanCount < 0),
+            TotalProducts = sum(ScanCount),
+            DepartmentCount = n_distinct(DepartmentDescription),
+            FinelineCount = n_distinct(FinelineNumber)
+            )
+# prop.table(table(tmp$Day))
+# prop.table(table(data$Day))
+head(data)
+
+dataDepartments = import %>%
+  group_by(TripType, VisitNumber, DepartmentDescription) %>%
+  summarise(count = n()) %>%
+  spread(DepartmentDescription, count, fill=0)
+head(dataDepartments)
+
+prodsPerDept = import %>%
+  group_by(VisitNumber, DepartmentDescription) %>%
+  summarise(
+    BoughtPerDepartment = sum(ScanCount >=0),
+    ReturnedPerDepartment = sum(ScanCount < 0)
+  )
+head(prodsPerDept)
+
+
+merged.data <- merge(data, dataDepartments, by=c("TripType", "VisitNumber"))
+merged.data <- merge(merged.data, prodsPerDept, by=c("VisitNumber"))
+merged.data <- subset(merged.data, select = -c(DepartmentDescription))
+```
+
+
+#Data Preparation
+Since the dataset is very complex now (with 80 variables and over 300k observations) I need to make some data preparations.
+
+Firstly I decrease the number of observations to 100k and then I will remove the `nearZeroVariance` variables.
+When the dataset is ready I split it into a training and testing set.
+
+```{r, echo=TRUE, message=FALSE, warning=FALSE}
+indxSample <- sample(seq_len(nrow(merged.data)), size = 100000)
+sample <- merged.data[indxSample, ]
+
+prop.table(table(sample$TripType))
+
+nzv <- nearZeroVar(sample)
+names(sample)[nzv]
+data <- sample[, -nzv]
+
+data$TripType <- factor(data$TripType)
+levels(data$TripType)
+
+names(data) = make.names(names(data), unique = FALSE)
+
+set.seed(1313)
+indxTrain <- createDataPartition(y = data$TripType, p = 0.75)
+train <- data[indxTrain$Resample1,]
+test <- data[-indxTrain$Resample1,]
+```
+
+
+#Classification
+##Random Forest
+```{r, echo=TRUE, message=FALSE, warning=FALSE, fig.align='center'}
+rf <- randomForest(TripType ~ ., data=train)
```

Comment:
The input data should be without "VisitNumber". It is not the predictor/ feature to classify the data. It is only an id of     any visit.